tests/base.py

```diff
@@ -29,11 +39,33 @@ def handle_toolbar_created(sender, toolbar=None, **kwargs):
         return response
 
 
+class AsyncToolbarTestClient(AsyncClient):
+    async def request(self, **request):
+        # Use a thread/async task context-local variable to guard against a
+        # concurrent _created signal from a different thread/task.
```

Comment:
Can you expand the comment to explain when this happens? E.g. when running the test suite using multiple threads or something?

Reply:
Ah I thought so. In a way outdated or confusing comments are worse than no comments at all. It would certainly help a lot if you could clarify this!